overlord/snapstate/component.go

```diff
@@ -61,11 +69,21 @@ func InstallComponents(ctx context.Context, st *state.State, names []string, inf
 		}
 	}
 
-	compsups, err := componentSetupsForInstall(ctx, st, names, snapst, snapst.Current, snapst.TrackingChannel, opts)
+	if vsets == nil {
+		vsets = snapasserts.NewValidationSets()
```

Comment:
shouldn't this be the enforced set?

Reply:
Good catch, my bad. This change needs tests anyways, my mistake.